File: rust/src/api/desktop_media.rs
use anyhow::Result;

use crate::desktop_media;

#[flutter_rust_bridge::frb]
pub fn desktop_ocr_image(
    bytes: Vec<u8>,
    language_hints: String,
) -> Result<desktop_media::ocr::OcrPayload> {
    desktop_media::ocr::desktop_ocr_image(&bytes, &language_hints)
}

#[flutter_rust_bridge::frb]
pub fn desktop_ocr_pdf(
    bytes: Vec<u8>,
    max_pages: u32,
    dpi: u32,
    language_hints: String,
) -> Result<desktop_media::ocr::OcrPayload> {
    desktop_media::ocr::desktop_ocr_pdf(&bytes, max_pages, dpi, &language_hints)
}

File: lib/core/runtime_compat/api/desktop_media.dart
import '../desktop_media/ocr.dart';

Future<OcrPayload> desktopOcrImage(
        {required List<int> bytes, required String languageHints}) =>
    throw UnsupportedError('rust_runtime_removed:desktopOcrImage');

Future<OcrPayload> desktopOcrPdf(
        {required List<int> bytes,
        required int maxPages,
        required int dpi,
        required int startPage,
        required String languageHints}) =>
    throw UnsupportedError('rust_runtime_removed:desktopOcrPdf');
